### ISSUE TITLE
ServiceStack.Text Json deserialization fails when a config scope TypeAttr is different than the global configuration

### ISSUE THREAD

**Issue by tylerohlsen** [ServiceStack/Issues#742]
When the global configuration `TypeAttr` is different than a scoped configuration `TypeAttr`, deserialzation while in the scope fails.  Here's a unit test that fails on the last line that demonstrates the issue:

```csharp
[Test]
public void ServiceStack_AllowRuntimeType()
{
    // Initialize static delegate to allow all types to be deserialized with the type attribute
    JsConfig.AllowRuntimeType = _ => true;
    JsConfig.TypeAttr = "$type";
    var example = new Example { Property = new MyProperty { Value = "Hello serializer" } };

    var serialized = JsonSerializer.SerializeToString(example);
    var deserialized = JsonSerializer.DeserializeFromString<Example>(serialized);
    Assert.IsNotNull(deserialized?.Property);

    // Now the same process with a config scope that has a TypeAttr that differs from the global TypeAttr value
    using var scope = JsConfig.With(new Config{TypeAttr = "_type"});
    serialized = JsonSerializer.SerializeToString(example);
    deserialized = JsonSerializer.DeserializeFromString<Example>(serialized);
    Assert.IsNotNull(deserialized?.Property);
}

private class Example
{
    public IProperty Property { get; set; }
}
private interface IProperty
{
}
private class MyProperty : IProperty
{
    public string Value { get; set; }
}
```

**Comment by mythz** [comment]
This should now be resolved from [this commit](https://github.com/ServiceStack/ServiceStack.Text/commit/4708869691e85b80e29d8ac0c99eac1345ca3883).

This change is available from the latest v5.9.3 that's now [available on MyGet](https://docs.servicestack.net/myget).

Thanks for the repro!

**Comment by tylerohlsen** [comment]
Awesome!  Thanks for getting this fixed so quickly!
